include/nbl/video/IGPUDescriptorSet.h

```diff
@@ -87,14 +87,14 @@ class IGPUDescriptorSet : public asset::IDescriptorSet<const IGPUDescriptorSetLa
 	private:
         inline void incrementVersion() { m_version.fetch_add(1ull); }
 
+        using redirect_t = IGPUDescriptorSetLayout::CBindingRedirect;
         friend class ILogicalDevice;
-        asset::IDescriptor::E_TYPE validateWrite(const IGPUDescriptorSet::SWriteDescriptorSet& write) const;
-        void processWrite(const IGPUDescriptorSet::SWriteDescriptorSet& write, const asset::IDescriptor::E_TYPE type);
-        bool validateCopy(const IGPUDescriptorSet::SCopyDescriptorSet& copy) const;
-        void processCopy(const IGPUDescriptorSet::SCopyDescriptorSet& copy);
+        asset::IDescriptor::E_TYPE validateWrite(const IGPUDescriptorSet::SWriteDescriptorSet& write, redirect_t::storage_range_index_t& descriptorRedirectBindingIndex, redirect_t::storage_range_index_t& mutableSamplerRedirectBindingIndex) const;
```

Comment:
instead of doing 1 return value + multiple out pointers or references, why not wrap all 3 into a struct? e.g.
```cpp
struct SValidationResult
{
   asset::IDescriptor::E_TYPE type;
   redirect_t::storage_range_index_t descriptorRedirect;
   redirect_t::storage_range_index_t mutableSamplerRedirect;
};
```

and pass this by `const&` to the `processWrite` ?

Also oa similar thing for the copies.